databaseutil/databaseutil.go

```diff
@@ -65,6 +66,48 @@ func InsertIntoUsersTable(
 	return nil
 }
 
+
+func InsertIntoNoteTable(
+	userId int64,
+	noteType string,
+	content string,
+	publicationId int64,
+	creationTime time.Time,
+) error {
+
+	sqlQuery := `
+		INSERT INTO notes (author_id, type, content, publication_id, creation_time)
+		VALUES ($1, $2, $3, $4, $5)`
+
+
+	var cleanPublicationId sql.NullInt64
+	var cleanNoteType sql.NullString
+
+	if publicationId < 1 {
```

Comment:
So go doesn't have a builtin nullable int type. 

options and why i didn't choose them
1) publicationId should be a nullable int in the note.go file? 
If so i don't like that option since nullable int a a sql exclusive object, and i wanted to keep our datasystem storage agnostic. 

2) why doesn't this function just accept a nullabelint instead of an int. Because you can't just assign pass in a int to a sql.NullableInt you have to create the struct you see in the next line. So that would mean the calling function would need to create the struct which i don't like.

Reply:
oh okay i didn't realize we couldn't do nullable int. I thought golang allowed something like:
```
var nullableNumber : int | null
```
or something like that

Reply:
So i agree our note struct shouldn't necessarily depend on a sql specific struct, but it should also be an accurate representation of our model. A 0 or -1 for publicationId is just not the same thing as null. One option would be to create our own concept of nullableInt

Reply:
Another option that I've been thinking about (but would involve a larger discussion) is to make our database rows immutable. Meaning notes wouldn't have publicationId on them, but there would be a separate association table relating noteIds to publicationIds

Reply:
this would kind of avoid the problem altogether. we would retrieve notes with all non-nullable fields, and we would retrieve a separate thing that would look like Map<noteId, publicationId>

Reply:
I agree with having a reference table, <NoteId, PublicationId> Table, and not storing publicationId in the table! Lets do that. Want me to do that as part of this change? or make a seperate pull request for that agree on that. Then come back to this?